xcube/core/subsampling.py

```diff
@@ -109,6 +114,37 @@ def subsample_dataset(
     return xr.Dataset(data_vars=new_data_vars, attrs=dataset.attrs)
 
 
+def _mode(var: xr.DataArray, x_name: str, y_name: str, step: int):
+    dim = dict()
+    drop_axis = []
+    if x_name in var.dims:
+        dim[x_name] = step
+        drop_axis.append(var.dims.index(x_name))
+    if y_name in var.dims:
+        dim[y_name] = step
+        drop_axis.append(var.dims.index(y_name))
+    var_coarsen = var.coarsen(dim=dim, boundary="pad", coord_func="min")
+    if drop_axis[0] > drop_axis[1]:
+        drop_axis[0] += 2
+        drop_axis[1] += 1
+    else:
+        drop_axis[0] += 1
+        drop_axis[1] += 2
```

Comment:
Extract helper function `_get_drop_axis()` that just computes `drop_axis` from `var`, `x_name`, `y_name`. The remaining code is a duplication of the first lines of new helper `_agg_builtin()`, see above. You can now extract from both `_agg_mode()` and `_agg_builtin()` common:

```
def _coarsen(vat, step, x_name, y_name, step):
  dim = dict()
  if x_name in var.dims:
      dim[x_name] = step
  if y_name in var.dims:
      dim[y_name] = step
  return var.coarsen(dim=dim, boundary="pad", coord_func="min")
```

Reply:
Also explain the magic numbers `1`, `2` used for drop_axis. They are not clear to me.